servers/parser-graphql-wrapper/src/databases/pocketMetadataStore.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  DynamoDBDocumentClient,
+  PutCommand,
+  QueryCommand,
+} from '@aws-sdk/lib-dynamodb';
+import config from '../config';
+import * as Sentry from '@sentry/node';
+import {
+  PocketMetadata,
+  PocketMetadataSource,
+} from '../__generated__/resolvers-types';
+import md5 from 'md5';
+
+export interface IPocketMetadataDataStore {
+  getStoredPocketMetadata(
+    resolvedUrl: string,
+    version: number,
+    source: PocketMetadataSource,
+  ): Promise<PocketMetadataEntity | null>;
+  storePocketMetadata(
+    input: PocketMetadataEntity,
+    ttl: number,
+  ): Promise<PocketMetadataEntity>;
+}
+
+export type PocketMetadataEntity = Omit<
+  PocketMetadata,
+  'createdAt' | 'datePublished'
+> & {
+  urlHash: string; // md5 hash of the resolved Url
+  createdAt: number; // epoch time in seconds
+  datePublished: number; // epoch time in seconds
+  version: number;
+  __typename: string;
+};
+
+export class ItemSummaryDataStoreBase implements IPocketMetadataDataStore {
+  public static table = config.dynamoDb.itemSummaryTable;
+
+  constructor(public conn: DynamoDBDocumentClient) {}
+  /**
+   * Fetch a share by ID from DynamoDB table
+   * @param id the lookup ID (hash key)
+   * @returns ShareEntity, or null if record does not exist
+   */
+  async getStoredPocketMetadata(
+    resolvedUrl: string,
+    version: number,
+    source: PocketMetadataSource,
+  ): Promise<PocketMetadataEntity | null> {
+    const getPreviewCommand = new QueryCommand({
+      TableName: ItemSummaryDataStoreBase.table.name,
```

Comment:
confused about this query -- what's the PK in the dynamo table?

Reply:
because you should be running a GetItem command, not a query command, I think...

Reply:
My understanding is you can't filter expression on a GetItem?

Reply:
PK is the hash of the URL